tests/test_buf_mut.rs

```diff
@@ -121,6 +122,41 @@ fn test_slice_put_bytes() {
     assert_eq!(&[17, 19, 19, 0], &v[..]);
 }
 
+#[test]
+fn test_uninit_buf_mut() {
+    let mut buf = [b'X'; 100];
+    let fill = [b'Z'; 100];
+
+    fn make(slice: &mut [u8]) -> &mut [MaybeUninit<u8>] {
+        // SAFETY: [T] and [MaybeUninit<T>] have the same layout.
+        unsafe { transmute(slice) }
+    }
+
+    {
+        let mut slice = make(&mut buf[..]);
+        slice.put_bytes(b'A', 2);
+        slice.put_slice(b"BBCC");
+        assert_eq!(b"AABBCCXX", &buf[..8]);
+    }
+
+    for buf_len in 0..100 {
+        for fill_len in 0..=buf_len {
+            let mut slice = make(&mut buf[..buf_len]);
+            slice.put_slice(&fill[..fill_len]);
+            assert_eq!(buf_len - fill_len, slice.remaining_mut());
+        }
+    }
+}
+
+#[test]
+#[should_panic]
+fn test_uninit_buf_mut_overflow() {
+    let mut buf = [b'X'; 4];
+    // SAFETY: [T] and [MaybeUninit<T>] have the same layout.
+    let mut slice: &mut [MaybeUninit<u8>] = unsafe { transmute(&mut buf[..]) };
+    slice.put_slice(b"AABBCC");
+}
```

Comment:
Let's use a slice of size `+1` rather than `+2` for the test.
```suggestion
#[test]
#[should_panic]
fn test_uninit_buf_mut_overflow() {
    let mut buf = [b'X'; 4];
    // SAFETY: [T] and [MaybeUninit<T>] have the same layout.
    let mut slice: &mut [MaybeUninit<u8>] = unsafe { transmute(&mut buf[..]) };
    slice.put_slice(&[b'Y'; 5]);
}
```